server/routes/api/prospects.js

```diff
@@ -24,170 +24,181 @@ const upload = multer({ storage: storage })
 // @route POST /api/prospects
 // @desc Create a Prospect object
 // @access Authenticated Users
-router.post("/", (req, res) => {
-  //Prospect field validation
-  const { errors, isValid } = validateProspectInput(req.body);
-
-    if (!isValid) {
-        return res.status(400).json(errors);
-    }
-    Prospect.findOne({ email: req.body.email }).then(prospect => {
-        if (prospect) {
-          return res.status(400).json({ email: "Prospect with this email address already exists" });
-        } else {
-          if(req.body.ownedBy !== null && !mongoose.Types.ObjectId.isValid(req.body.ownedBy)){
-            return res.status(400).json({ ownedBy: "Invalid user id provided for ownedBy"});
-          }
-          const {firstName, lastName, email, ownedBy, status} = req.body;
-          const newProspect = new Prospect({firstName, lastName, email, ownedBy, status});
-    newProspect
-        .save()
-        .then(prospect => res.json(prospect))
-        .catch(err => console.log(err));
-    }
-  });
+router.post("/", passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    //Prospect field validation
+    const ownedBy = req.user.id;
+    const { errors, isValid } = validateProspectInput(req.body);
+
+      if (!isValid) {
+          return res.status(400).json(errors);
+      }
+      Prospect.findOne({ email: req.body.email }).then(prospect => {
+          if (prospect) {
+            return res.status(400).json({ email: "Prospect with this email address already exists" });
+          } else {
+            if(ownedBy !== null && !mongoose.Types.ObjectId.isValid(ownedBy)){
```

Comment:
For all of these routes, you should check if ownedBy is the same as req.user.id, or else you shouldn't be able to get, edit, or anything with that prospect